File: frontend/src/types/notification.ts
export type NotificationType = 
  | 'DROP_ANNOUNCED'
  | 'DROP_OPEN'
  | 'DROP_CLOSING_SOON'
  | 'ORDER_CONFIRMED'
  | 'ORDER_READY'
  | 'ORDER_CANCELLED'
  | 'NEW_FOLLOWER'
  | 'LOW_STOCK';

export interface NotificationResponse {
  notificationId: number;
  type: NotificationType;
  title: string;
  message: string;
  referenceType: 'DROP' | 'ORDER' | 'USER' | null;
  referenceId: number | null;
  isRead: boolean;
  createdAt: string;
  timeAgo: string;
}

File: backend/src/main/java/com/foodflow/model/Notification.java
package com.foodflow.model;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;
import lombok.Builder;
import org.hibernate.annotations.CreationTimestamp;

import java.time.LocalDateTime;

@Entity
@Table(name = "notifications")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Notification {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long notificationId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @Enumerated(EnumType.STRING)
    private NotificationType type;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String message;

    @Enumerated(EnumType.STRING)
    private ReferenceType referenceType;

    private Long referenceId;

    @Column(unique = true)
    private String eventKey;

    @Builder.Default
    private Boolean isRead = false;

    @CreationTimestamp
    private LocalDateTime createdAt;

    public enum NotificationType {
        DROP_ANNOUNCED, DROP_OPEN,
        ORDER_CONFIRMED, ORDER_READY, ORDER_CANCELLED,
        NEW_FOLLOWER, LOW_STOCK
    }

    public enum ReferenceType {
        DROP, ORDER, USER, NONE
    }
}
